### PR TITLE
fix(server): properly display next page object

### PR DESCRIPTION
**Pull request by bjgiraudon**
### Description
This PR fixes the `next` link display in `search` requests (both `GET` and `POST`).
It provides a URL to the next page and the json body in the case of a `POST` request.

### CODE REVIEW

**Review by alambare-csgroup — APPROVED** [review]

File: eodag/rest/utils.py [hunk 1]
```diff
@@ -744,13 +752,21 @@ def search_stac_items(url, arguments, root="/", catalogs=[], provider=None):
...
+    :param method: (optional) search request HTTP method ('GET' or 'POST')
+    :type method: str
     :returns: Catalog dictionnary
     :rtype: dict
     """
     collections = arguments.get("collections", None)
 
     catalog_url = url.replace("/items", "")
 
+    query_args = {
```
Comment (sbrunato) [inline eodag/rest/utils.py:764]: please rename `query_args` to `next_page_kwargs`

File: eodag/rest/utils.py [hunk 2]
```diff
@@ -731,7 +732,14 @@ def get_stac_catalogs(url, root="/", catalogs=[], provider=None, fetch_providers
     ).get_stac_catalog()
 
 
-def search_stac_items(url, arguments, root="/", catalogs=[], provider=None):
+def search_stac_items(
```
Comment (sbrunato) [inline eodag/rest/utils.py:735]: please rebase to get type hints from `develop` branch

**Review by sbrunato — CHANGES_REQUESTED** [review]
please add tests for GET & POST methods

File: eodag/rest/utils.py (hunk 2 above)
Comment (alambare-csgroup) [reply eodag/rest/utils.py]: done

**Review by sbrunato — APPROVED** [review]

**Pull request merged**